source/robot_model/tests/testRobotModel.cpp

```diff
@@ -1,8 +1,103 @@
 #include "robot_model/Model.hpp"
+
 #include <gtest/gtest.h>
 
+#include "robot_model/Exceptions/InvalidJointStateSizeException.hpp"
+#include "robot_model/Exceptions/FrameNotFoundException.hpp"
+
 using namespace RobotModel;
 
+class RobotModelTest : public testing::Test {
+protected:
+  void SetUp() override {
+    robot_name = "franka";
+    urdf = "/tmp/control_lib/robot_model/tests/panda_arm.urdf";
+    franka = Model(robot_name, urdf);
+    joint_state = StateRepresentation::JointState(robot_name, 7);
+  }
+
+  Model empty = Model();
+  Model franka;
+  std::string robot_name;
+  std::string urdf;
+
+  StateRepresentation::JointState joint_state;
+};
+
+TEST_F(RobotModelTest, TestInitEmptyModel) {
+  bool except_thrown = false;
```

Comment:
I dont know why but here I can't put `EXCPECT_THROW(empty.init_model(), std::invalid_argument)`. The expect_throw only seems to work with our own exceptions..

Reply:
It's probably because the exception object is different than just the default constructed `std::invalid_argument` - it has more context like "The file does not contain a valid URDF model", so the exception object does not match the expectation here.

Btw, I suggest you do
```c++
try {
   empty.init_model();
   FAIL() << "Expected exception when initialised empty model";
 } catch (const std::invalid_argument& e) {
 }
```

Reply:
Ah ignore the second part of my comment, there are a bunch of issues with that. Keep it how you have it.